src/search.cpp

```diff
@@ -114,6 +115,21 @@ setup_queryParser(Xapian::QueryParser* queryparser,
     }
 }
 
+// In suggestion mode, Parse subquery_phrase with OP_PHRASE as default op, set a window size equal to
+// the number of terms in the query. Combine this subquery_phrase with subquery_and and return it.
+Xapian::Query parseQuery(Xapian::QueryParser* queryParser, std::string qs, int flags, std::string prefix, bool suggestion_mode) {
+    Xapian::Query query, subquery_and;
+    query = subquery_and = queryParser->parse_query(qs, flags, prefix);
+
+    if (suggestion_mode) {
+      queryParser->set_default_op(Xapian::Query::op::OP_PHRASE);
+      Xapian::Query subquery_phrase = queryParser->parse_query(qs);
+      subquery_phrase = Xapian::Query(Xapian::Query::OP_PHRASE, subquery_phrase.get_terms_begin(), subquery_phrase.get_terms_end(), subquery_phrase.get_length());
+      query = Xapian::Query(Xapian::Query::OP_OR, subquery_phrase, subquery_and);
```

Comment:
How do you ensure that `subquery_phrase` has higher weight than `subquery_and`. Does it happen automatically due to some rules in Xapian?

Reply:
We are not providing higher weightage to `subquery_phrase`. The way this works is, suppose for a particular document, `subquery_and` gives a weightage A and `subquery_phrase` gives a weightage B. When we join these two queries with OP_OR, the net weight of this document becomes A+B. So this is why we are not losing any of the document which we used to get with only a single query. We are promoting the documents with terms in correct order with the added weight of `subquery_phrase`. We have to do it because just using `subquery_phrase` makes the search too exclusive.

I should perhaps add this as a comment in the code itself.